### PR TITLE
feat: added Quiz and Result Page

### PR DESCRIPTION
**Pull request by srv-pranjal**
# Quiz and Result Page

## Description

 - Added Rules Page
 - Created quiz context and loader context to load the data from API
 - Added document title hook
 - Added Quiz Page
 - Added Result Page
 
 ## Deploy Preview: [https://quizzers-git-quiz-result-srv-pranjal.vercel.app/](https://quizzers-git-quiz-result-srv-pranjal.vercel.app/)
 
image

image

image

### CODE REVIEW

File: src/components/CategoryCard/CategoryCard.js [hunk 1]
```diff
@@ -1,9 +1,16 @@
...
 import "./CategoryCard.css";
 
 export const CategoryCard = ({ categoryName, image }) => {
+  const { quizDispatch } = useQuiz();
   return (
-    <Link to="/rules">
+    <Link
+      to="/rules"
+      onClick={() =>
+        quizDispatch({ type: "SET_CATEGORY", payload: categoryName })
```
Comment (Namrata-J) [inline src/components/CategoryCard/CategoryCard.js:11]: could use constants for types!!(just a suggestion)

File: src/components/index.js [hunk 2]
```diff
@@ -1,3 +1,4 @@
 export { Navbar } from "./Navbar/Navbar";
```
Comment (Namrata-J) [inline src/components/index.js:1]: its great that you are using barrel exports!

File: src/index.css [hunk 3]
```diff
@@ -14,12 +14,11 @@ body {
 :root {
   --overlay-color: rgba(0, 0, 0, 0.7);
   --primary-color: rgb(104, 255, 98);
-  --light-primary-color: #ffd35a;
-  --dark-primary-color: #cca027;
   --grey-color: #aaaaaa;
   --background-color: #1f1d2b;
   --dark-background-color: #1a1922;
   --transparent-color: rgba(0, 0, 0, 0.1);
```
Comment (Namrata-J) [inline src/index.css:20]: either use hex or rgba to be consistent!!

**Review by Namrata-J — APPROVED** [review]
rest odf the code looks modular!

File: src/contexts/quiz-context.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+            };
+          });
+          quizDispatch({ type: "SET_QUESTIONS", payload: questionsArray });
+          quizDispatch({ type: "RESET_SCORE" });
+          quizDispatch({ type: "RESET_SELECTED_OPTIONS" });
+        }
+      } catch (error) {
+      } finally {
+        setShowLoader(false);
+      }
```
Comment (dev-enforced) [inline src/contexts/quiz-context.js:53]: Got an idea of implementing Loaders for the project through try catch method. Good work @srv-pranjal

File: src/pages/QuizQuestion/QuizQuestion.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+          {options.map((option, index) => {
+            return (
+              <button
+                key={index}
+                className={classNames("quiz__option", {
+                  "quiz__option--incorrect":
+                    selectedOption === option && option !== correctAnswer,
+                  "quiz__option--correct":
+                    selectedOption && option === correctAnswer,
+                })}
```
Comment (dev-enforced) [inline src/pages/QuizQuestion/QuizQuestion.js:48]: This is also new for me. Will go through the package of `classNames`

**Review by chanchal-panpaliya — APPROVED** [review]
code looks clean with proper naming with good UI

File: src/components/CategoryCard/CategoryCard.js (hunk 1 above)
Comment (srv-pranjal) [reply src/components/CategoryCard/CategoryCard.js:11]: Sure. Thanks for the suggestion.

File: src/index.css (hunk 3 above)
Comment (srv-pranjal) [reply src/index.css:20]: Sure. Thanks for the suggestion

**Review by dev-enforced — APPROVED** [review]
The code looks good and well structured. Good work @srv-pranjal on the absolute imports part.
Do check out the live preview once especially the questions related part as I was receiving errors in receiving the questions of a particular quiz after moving from rules

File: src/contexts/quiz-context.js (hunk 4 above)
Comment (srv-pranjal) [reply src/contexts/quiz-context.js:53]: Thanks for pointing out. It was an issue related to environment variables.
Fixed it.

**Pull request merged**